src/bdfsclient/ActionHandler.cpp

```diff
@@ -78,8 +93,38 @@ namespace dfs
 		return 0;
 	}
 
+  void ActionHandler::Unmount(std::string nbdPath, bool matchAll=false)
+  {
+    std::ifstream fp("/proc/mounts");
+    std::string line,s1,s2;
+    while(fp >> s1 >> s2 && std::getline(fp,line))
+    {
+      if((s1.compare(0,nbdPath.length(),nbdPath) == 0 && matchAll) 
+        || s1 == nbdPath)
+      {
+        printf("Unmounting %s\n",s2.c_str());
+        std::string cmd = "umount " + s2;
+        system(cmd.c_str());
+      }
+    }
+    fp.close();
+  }
+
+  void ActionHandler::Cleanup()
+  {
+    printf("Unmounting volumes...\n");
+    ActionHandler::Unmount("/dev/nbd",true);
+
+    for(auto &it : volumeInfo)
+    {
+      printf("Cleaning %s...\n",it.second->nbdPath.c_str());
+      ubd_disconnect(it.second->nbdPath.c_str());
+      delete it.second;
+    }
+  }
+
   // returns {0: fail, 1: success, 2: already binded}
-  int ActionHandler::BindVolume(const std::string name)
+  int ActionHandler::BindVolume(const std::string name, const std::string path)
```

Comment:
A common practice is to have argument as either const reference or non-const values. const value argument brings no benefit but restrictions (which usually may cause extra copies if that's not your original intension)